src/main/java/core/basesyntax/serviceimpl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportGenerator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ReportGeneratorImpl implements ReportGenerator<List<String>> {
+    @Override
+    public List<String> getReport() {
+        Map<String, Integer> fruitStorage = Storage.fruitStorage;
+        return fruitStorage.entrySet()
+                .stream()
+                .map(entry -> entry.getKey() + "," + entry.getValue())
+                .collect(Collectors.toList());
+    }
+}
```

Comment:
The class `ReportGeneratorImpl` is directly accessing the `Storage.fruitStorage` which can lead to high coupling between the service layer and the storage class. Consider using dependency injection to pass the necessary data to the `ReportGeneratorImpl` class, making it more independent and in line with SOLID principles.